drivers/storage/portworx/component/grafana.go

```diff
@@ -0,0 +1,430 @@
+				ConfigMap: &v1.ConfigMapVolumeSource{
+					LocalObjectReference: v1.LocalObjectReference{
+						Name: PXGrafanaDashboardsJsonConfigMap,
+					},
+				},
+			},
+		},
+	}
+)
+
+type grafana struct {
+	k8sClient        client.Client
+	scheme           *runtime.Scheme
+	recorder         record.EventRecorder
+	isGrafanaCreated bool
+}
+
+func (c *grafana) Name() string {
+	return GrafanaComponentName
+}
+
+func (c *grafana) Priority() int32 {
+	return DefaultComponentPriority
+}
+
+func (c *grafana) Initialize(
+	k8sClient client.Client,
+	_ version.Version,
+	scheme *runtime.Scheme,
+	recorder record.EventRecorder,
+) {
+	c.k8sClient = k8sClient
+	c.scheme = scheme
+	c.recorder = recorder
+}
+
+func (c *grafana) IsPausedForMigration(cluster *corev1.StorageCluster) bool {
+	return util.ComponentsPausedForMigration(cluster)
+}
+
+func (c *grafana) IsEnabled(cluster *corev1.StorageCluster) bool {
+	return cluster.Spec.Monitoring != nil &&
+		cluster.Spec.Monitoring.Grafana != nil &&
+		cluster.Spec.Monitoring.Grafana.Enabled &&
+		cluster.Spec.Monitoring.Prometheus != nil &&
+		cluster.Spec.Monitoring.Prometheus.Enabled
+}
+
+func (c *grafana) Reconcile(cluster *corev1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+
+	if err := c.createConfigmapFromFiles([]string{pxGrafanaDashboardConfig}, PXGrafanaDashboardConfigConfigMap, cluster.Namespace, ownerRef); err != nil {
+		return err
+	}
+	if err := c.createConfigmapFromFiles([]string{pxGrafanaDatasource}, PXGrafanaDatasourceConfigConfigMap, cluster.Namespace, ownerRef); err != nil {
+		return err
+	}
+	if err := c.createConfigmapFromFiles([]string{pxClusterDashboard, pxEtcdDashboard, pxNodeDashboard, pxPerformanceDashboard, pxVolumeDashboard},
+		PXGrafanaDashboardsJsonConfigMap, cluster.Namespace, ownerRef); err != nil {
+		return err
+	}
+	if err := c.createGrafanaDeployment(cluster, ownerRef); err != nil {
+		return err
+	}
+	if err := c.createGrafanaService(cluster, ownerRef); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *grafana) Delete(cluster *corev1.StorageCluster) error {
+	ownerRef := metav1.NewControllerRef(cluster, pxutil.StorageClusterKind())
+	err := k8sutil.DeleteService(c.k8sClient, GrafanaServiceName, cluster.Namespace, *ownerRef)
+	if err != nil {
+		return err
+	}
+	err = k8sutil.DeleteDeployment(c.k8sClient, GrafanaDeploymentName, cluster.Namespace, *ownerRef)
+	if err != nil {
+		return err
+	}
+	err = k8sutil.DeleteConfigMap(c.k8sClient, PXGrafanaDashboardConfigConfigMap, cluster.Namespace, *ownerRef)
+	if err != nil {
+		return err
+	}
+	err = k8sutil.DeleteConfigMap(c.k8sClient, PXGrafanaDashboardsJsonConfigMap, cluster.Namespace, *ownerRef)
+	if err != nil {
+		return err
+	}
+	err = k8sutil.DeleteConfigMap(c.k8sClient, PXGrafanaDatasourceConfigConfigMap, cluster.Namespace, *ownerRef)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *grafana) MarkDeleted() {}
```

Comment:
Is this intentionally empty? If so, can we add a comment as to why this is okay/what it's "not doing"?

Reply:
Yeah, most components have this as empty. I've added a comment